snowflake/custom-recipes/python-custom-recipes/recipe.py

```diff
@@ -1,113 +1,173 @@
 import os
 import sys
 import json
+import re
 import dataiku
 import urllib.parse as urlparse
 import snowflake.connector as sf
 from dataiku.customrecipe import *
+from dataiku.core.intercom import backend_json_call
 
 # Settings
 DATASET_IN     = get_input_names_for_role("input_dataset")[0]
 DATASET_OUT    = get_output_names_for_role("output_dataset")[0]
+APPEND_NOT_OVERWRITE = get_recipe_config().get("append_instead_of_overwrite", False)
+USE_INPUT_CONNECTION = get_recipe_config().get("use_input_connection", False)
+USE_PROJECT_VARIABLES = get_recipe_config().get("use_project_variables", False)
 
-AWS_ACCESS_KEY = get_recipe_config().get("aws_access_key")
-AWS_SECRET_KEY = get_recipe_config().get("aws_secret_key")
+AWS_ACCESS_KEY = None
+AWS_SECRET_KEY = None
 
-if not AWS_ACCESS_KEY or not AWS_SECRET_KEY:
-    # Looking up in Project Variables
-    print("[+] AWS Access Key or Secret Key not entered in the Plugin interface. Looking into Project Variables...")    
+# Dataiku Datasets
+ds = dataiku.Dataset(DATASET_IN)
+out = dataiku.Dataset(DATASET_OUT)
+
+# Dataset configurations
+config_in = ds.get_config()
+config_out = out.get_location_info(sensitive_info=True)["info"]
+
+
+if USE_INPUT_CONNECTION:
+    print("[+] Use the S3 credentials in the Input's connection ({})".format(DATASET_IN))
+    connection_name = config_in["params"]["connection"]
+    input_connection = backend_json_call("connections/get-details", data={
```

Comment:
Understood this a workaround to a DSS bug with the get_location_info() when the dataset is S3, unfortunately it won't work unless the plugin user is a DSS admin, which is a show stopper.
PR is put on hold till the bug is resolved in next DSS releases